backend/src/service/south.service.ts

```diff
@@ -531,6 +533,166 @@ export default class SouthService {
+
+    if (Array.isArray(settings[arrayKey])) {
+      arrayData = (settings[arrayKey] as Array<unknown>).filter(
+        (item): item is Record<string, unknown> => typeof item === 'object' && item !== null
+      );
+    } else {
+      const items = this.listItems(southId);
+      const allArrayData: Array<Record<string, unknown>> = [];
+
+      for (const item of items) {
+        const itemSettings = item.settings as unknown as Record<string, unknown>;
+        if (Array.isArray(itemSettings[arrayKey])) {
+          allArrayData.push(
+            ...(itemSettings[arrayKey] as Array<unknown>).filter(
+              (item: unknown): item is Record<string, unknown> => typeof item === 'object' && item !== null
+            )
+          );
+        }
+      }
+      arrayData = allArrayData;
+    }
+
+    return arrayData;
+  }
+
+  async checkArrayCSVImport(
+    file: Express.Multer.File,
+    delimiter: string,
+    arrayKey: string,
+    southType: OIBusSouthType,
+    existingItems: Array<Record<string, unknown>> = []
+  ): Promise<{
+    items: Array<Record<string, unknown>>;
+    errors: Array<{ item: Record<string, string>; error: string }>;
+  }> {
+    if (!file) {
+      throw new OIBusValidationError('File is null or undefined');
+    }
+
+    let fileContent: string;
+    if (file.buffer) {
+      fileContent = file.buffer.toString('utf8');
+    } else if (file.path) {
+      fileContent = (await fs.readFile(file.path)).toString('utf8');
+    } else {
+      throw new OIBusValidationError(
+        `File has neither buffer nor path. File object: ${JSON.stringify({ fieldname: file.fieldname, originalname: file.originalname, mimetype: file.mimetype })}`
+      );
+    }
+
+    if (!fileContent || fileContent.trim().length === 0) {
+      throw new OIBusValidationError('File content is empty');
+    }
+
+    const manifest = this.listManifest().find((southManifest: SouthConnectorManifest) => southManifest.id === southType);
+
+    if (!manifest) {
+      throw new NotFoundError(`South manifest "${southType}" not found`);
+    }
+
+    const arrayAttribute = this.getArrayAttributeFromManifest(manifest, arrayKey);
+
+    if (!arrayAttribute) {
+      throw new NotFoundError(`Array field "${arrayKey}" not found in manifest`);
+    }
+
+    return validateArrayCSVImport(fileContent, delimiter, arrayAttribute, existingItems);
+  }
+
+  async checkArrayFileImport(
+    southId: string,
+    file: Express.Multer.File,
+    delimiter: string,
+    arrayKey: string
+  ): Promise<{
+    items: Array<Record<string, unknown>>;
+    errors: Array<{ item: Record<string, string>; error: string }>;
+  }> {
+    if (!file) {
+      throw new OIBusValidationError('Missing file');
+    }
+    if (!delimiter) {
+      throw new OIBusValidationError('Missing delimiter');
+    }
+
+    const southConnector = this.findById(southId);
+    const existingItems = this.getArrayFieldItemsFromDatabase(southId, arrayKey);
```

Comment:
let's check if we are in create mode (in this case, no need to retrieve from database)